src/main/java/com/epam/service/TraineeService.java

```diff
@@ -7,8 +7,13 @@
 public interface TraineeService {
 
     void createTrainee(Trainee trainee);
-    void updateTrainee(Trainee trainee);
+    boolean updateTrainee(Trainee trainee);
     void deleteTrainee(Integer traineeId);
```

Comment:
This comment is applicable to all of your classes in this `service` package.
I would recommend to change the return types to `Trainee` or `Optional<Trainee>`. This is a very handy technique which could help you to avoid additional DB calls. Imagine that you are updating your entity and then you need to do something with updated data, e.g. compute the diff between the old entity and updated entity, and save the diff to the audit log table. So, returning the updated/created or even deleted entity could be useful. And this is what the Spring Data JPA repositories provide us:
```java
<S extends T> S save(S entity);
<S extends T> Iterable<S> save(Iterable<S> entities);
```